packages/mdc-select/index.ts

```diff
@@ -0,0 +1,596 @@
+
+    if (this.root_.classList.contains(cssClasses.DISABLED)
+      || (this.nativeControl_ && this.nativeControl_.disabled)) {
+      this.disabled = true;
+    }
+  }
+
+  destroy() {
+    // One of these elements must be non-null.
+    const element = this.nativeControl_ || this.selectedText_!;
+
+    element.removeEventListener('change', this.handleChange_);
+    element.removeEventListener('focus', this.handleFocus_);
+    element.removeEventListener('blur', this.handleBlur_);
+    element.removeEventListener('keydown', this.handleKeydown_);
+    MOUSEDOWN_TOUCHSTART_EVENTS.forEach((evtType) => {
+      element.removeEventListener(evtType, this.handleClick_ as EventListener);
+    });
+
+    if (this.menu_) {
+      this.menu_.unlisten(menuSurfaceConstants.strings.CLOSED_EVENT, this.handleMenuClosed_);
+      this.menu_.unlisten(menuSurfaceConstants.strings.OPENED_EVENT, this.handleMenuOpened_);
+      this.menu_.unlisten(menuConstants.strings.SELECTED_EVENT, this.handleMenuSelected_);
+      this.menu_.destroy();
+    }
+
+    if (this.ripple) {
+      this.ripple.destroy();
+    }
+    if (this.outline_) {
+      this.outline_.destroy();
+    }
+    if (this.leadingIcon_) {
+      this.leadingIcon_.destroy();
+    }
+    if (this.helperText_) {
+      this.helperText_.destroy();
+    }
+    if (this.validationObserver_) {
+      this.validationObserver_.disconnect();
+    }
+
+    super.destroy();
+  }
+
+  getDefaultFoundation(): MDCSelectFoundation {
+    return new MDCSelectFoundation({
+          ...(this.nativeControl_ ? this.getNativeSelectAdapterMethods_() : this.getEnhancedSelectAdapterMethods_()),
+          ...this.getCommonAdapterMethods_(),
+          ...this.getOutlineAdapterMethods_(),
+          ...this.getLabelAdapterMethods_(),
+        },
+        this.getFoundationMap_(),
+    );
+  }
+
+  /**
+   * Handles setup for the enhanced menu.
+   */
+  private enhancedSelectSetup_(menuFactory: MenuFactory) {
+    const isDisabled = this.root_.classList.contains(cssClasses.DISABLED);
+    this.selectedText_!.setAttribute('tabindex', isDisabled ? '-1' : '0');
+    this.hiddenInput_ = this.root_.querySelector(strings.HIDDEN_INPUT_SELECTOR);
+    this.menuElement_ = this.root_.querySelector(strings.MENU_SELECTOR)!;
+    this.menu_ = menuFactory(this.menuElement_);
+    this.menu_.hoistMenuToBody();
+    this.menu_.setAnchorElement(this.root_);
+    this.menu_.setAnchorCorner(menuSurfaceConstants.Corner.BOTTOM_START);
+    this.menu_.wrapFocus = false;
+  }
+
+  private initRipple_(): MDCRipple {
+    // At least one of the elements must be non-null.
+    const element = this.nativeControl_ || this.selectedText_!;
+
+    // tslint:disable:object-literal-sort-keys
+    const foundation = new MDCRippleFoundation({
+      ...MDCRipple.createAdapter(this),
+      registerInteractionHandler: <E extends EventType>(evtType: E, handler: SpecificEventListener<E>) => {
+        element.addEventListener(evtType, handler);
```

Comment:
do we need a guard for these method calls too?

Reply:
`element` should never be null:

```ts
// One of these elements must be non-null.
const element = this.nativeControl_ || this.selectedText_!;
```